src/workerd/jsg/setup.h

```diff
@@ -414,23 +445,39 @@ class Isolate: public IsolateBase {
       v8::Isolate::CreateParams createParams = {})
       : Isolate(system, nullptr, kj::mv(observer), kj::mv(createParams)) {}
 
+  template <typename MetaConfiguration>
+  void instantiateDefaultWrapper(MetaConfiguration&& configuration) {
+    KJ_DASSERT(wrappers[0].get() == nullptr);
+    auto wrapper = wrapperSpace.construct(ptr, kj::fwd<MetaConfiguration>(configuration));
+    wrapper->initTypeWrapper();
+    wrappers[0] = kj::mv(wrapper);
+  }
+
+  template <typename MetaConfiguration>
+  kj::Own<TypeWrapper>& instantiateWrapper(MetaConfiguration&& configuration) {
+    hasExtraWrappers = true;
+    return wrappers.add(kj::heap<TypeWrapper>(ptr, kj::fwd<MetaConfiguration>(configuration)));
+  }
+
   ~Isolate() noexcept(false) {
-    dropWrappers(kj::mv(wrapper));
+    dropWrappers(wrappers.releaseAsArray());
```

Comment:
You could avoid the need to templatize/inline `dropWrappers()` by instead doing something here like:

```
int dummy;
dropWrappers(kj::attachRef(dummy, kj::mv(wrappers)));
```

Or alternatively, `dropWrappers()` could take a `kj::FunctionParam<void()> drop` which is invoked at the point where the wrappers should be dropped, and then this could be:

```
dropWrappers([&]() { wrappers.clear(); });
```

(Using `FunctionParam` for this avoids any memory allocation, but still permits `dropWrappers()` to be declared in the c++ file.)